File: nvxapp.client/src/nvx/ClientServer-Service/Account/Models/user-load-model.ts
import { ModelResult } from "../../ModelsBase/model-result";
import { RolesModel } from "./user-roles-model";


export class UserLoadInModel {
  public id: string | null;
}

export class UserLoadOutModel extends ModelResult {

  public userData: UserDataModel | null;

}

export class UserDataModel  {

  public id: string | null;

  public userName: string | null;

  public roles: RolesModel[] = [];

}

File: nvxapp.server.service/ClientServer-Service/Account/Models/UserLoadModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static nvxapp.server.data.Entities.AspNetUsersDataUtil;

namespace nvxapp.server.service.ClientServer_Service.Account.Models
{
    public class UserLoadInModel
    {
        public string? Id { get; set; }
    }

    public class UserLoadOutModel : ModelResult
    {
        public UserDataModel UserData { get; set; }
        
        public UserLoadOutModel(){
            UserData = new UserDataModel();
        }
    }

    public class UserDataModel 
    {
        public string? Id { get; set; }
        public string? UserName { get; set; }
        public List<AspNetRolesModel> Roles { get; set; }

        public UserDataModel()
        {
            Roles = new List<AspNetRolesModel>();
        }
    }

}
